### DISCUSSION TITLE
Editable Cycle highlight action

### DISCUSSION THREAD

**Discussion by pedrogmattos** [koreader/koreader#11778] [Ideas]
I like to have my thumb on the screen of the e-reader while reading, and for that I need to disable the hightlight action, but I also want to occasionally highlight text, so I have to enable highlight again.

At the moment I have one gesture set to `Set highlight action: Do nothing` and other to `Set highlight action: Highlight`. This works fine, but I rather have just one gesture to enable and disable highlight. I have tried to use `Cycle hightlight action`, but it cycles through many options before coming back to the one I want.

I would like have an editable `Cycle highlight action` so I can select only `Do nothing` and `Highlight`. I envision it as this: when you click on the option `Cycle highlight action` in the gestures menu, it takes you to a menu like the `Set highlight action` menu, where you can select all the available highlight options you wish that specific gesture to cycle through.

Is this easy to do?
(I do not know how to code, so I cannot attemp to do it myself, unfortunately. My contributions are restricted to translating KoReader...)

I figure this would also be useful for other people and other use cases, like toggling between Dictionary and Wikipedia, for instance.

**Comment by hius07** [comment]
Create a file named `2-hl-action-cycle.lua` with the following lines
```Lua
local ReaderHighlight = require("apps/reader/modules/readerhighlight")

ReaderHighlight.onCycleHighlightAction = function(self)
    local current_action = G_reader_settings:readSetting("default_highlight_action")
    self:onSetHighlightAction(current_action == "nothing" and 3 or 2)
    return true
end

```
put it to `koreader/patches` folder, restart.
Will cycle through "Do nothing" and "Highlight" only.
